sources/gateway_sui.move

```diff
@@ -122,20 +122,6 @@ entry fun withdraw_to_address<T>(gateway: &mut Gateway, amount:u64, nonce:u64, r
     transfer::public_transfer(coin, recipient);
 }
 
-
-// === View Functions ===
```

Comment:
A nonce getter could be helpful. The tx broadcaster can read nonce from gateway, and stops retrying if the latest nonce is already higher than the tx's nonce being broadcasted.

Reply:
You don't have view functions such as Solidity that returns value without side effects, all functions are called in the context of a transaction, to get data you would query the gateway object and read its data directly

Reply:
If `nonce` can be retrieved without this function, it would be fine. The Solana nonce is retrieved from PDA account info.

Reply:
You query the gateway object from its ID, then you can read the nonce in the gateway object